src/main/java/com/adoptmeplus/enterprise/dao/AdoptionDAO.java

```diff
@@ -21,8 +21,11 @@
 @Profile("dev")
 public class AdoptionDAO implements IAdoptionDAO{
 
-    @Autowired
-    private AdoptionRepository adoptionRepository;
+    private final AdoptionRepository adoptionRepository;
```

Comment:
Constructor injection instead of field injection: https://odrotbohm.de/2013/11/why-field-injection-is-evil/